### ISSUE TITLE
Kafka server fails when HDFS connector is running for long time with large flush.size

### ISSUE THREAD

**Issue by nivemaham** [confluentinc/kafka-connect-hdfs#152]
I am trying to run HDFS connector on a single-node cluster setting. When connector is started, initially it commits offsets. After a while i get this exception and kafka-server fails.
This is my HDFS connector properties.
```
name=radar-hdfs-sink-android-15000
connector.class=io.confluent.connect.hdfs.HdfsSinkConnector
tasks.max=4
topics=android_empatica_e4_electrodermal_activity,android_empatica_e4_inter_beat_interval,android_empatica_e4_temperature
flush.size=15000
hdfs.url=hdfs://localhost:9000
format.class=org.radarcns.sink.HDFS.AvroFormatRadar
topics.dir=topicAndroidNew
```
After a while i get this exception.
```
[2016-11-17 15:35:32,467] ERROR Task radar-hdfs-sink-android-test-2 threw an uncaught and unrecoverable exception (org.apache.kafka.connect.runtime.WorkerTask:142)
[java.lang.NullPointerException
    at org.apache.kafka.clients.consumer.internals.ConsumerNetworkClient.poll(ConsumerNetworkClient.java:162)
    at org.apache.kafka.clients.consumer.internals.AbstractCoordinator.ensureCoordinatorReady(AbstractCoordinator.java:181)
    at org.apache.kafka.clients.consumer.internals.AbstractCoordinator.ensureActiveGroup(AbstractCoordinator.java:229)
    at org.apache.kafka.clients.consumer.internals.ConsumerCoordinator.ensurePartitionAssignment(ConsumerCoordinator.java:366)
    at org.apache.kafka.clients.consumer.KafkaConsumer.pollOnce(KafkaConsumer.java:975)
    at org.apache.kafka.clients.consumer.KafkaConsumer.poll(KafkaConsumer.java:938)
    at org.apache.kafka.connect.runtime.WorkerSinkTask.pollConsumer(WorkerSinkTask.java:316)
    at org.apache.kafka.connect.runtime.WorkerSinkTask.poll(WorkerSinkTask.java:222)
    at org.apache.kafka.connect.runtime.WorkerSinkTask.iteration(WorkerSinkTask.java:170)
    at org.apache.kafka.connect.runtime.WorkerSinkTask.execute(WorkerSinkTask.java:142)
    at org.apache.kafka.connect.runtime.WorkerTask.doRun(WorkerTask.java:140)
    at org.apache.kafka.connect.runtime.WorkerTask.run(WorkerTask.java:175)
    at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
    at java.util.concurrent.FutureTask.run(FutureTask.java:266)
    at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
    at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
    at java.lang.Thread.run(Thread.java:745)
[2016-11-17 15:35:32,468] ERROR Task is being killed and will not recover until manually restarted (org.apache.kafka.connect.runtime.WorkerTask:143)
[2016-11-17 15:35:32,468] INFO Marking the coordinator radar-test.thehyve.net:9092 (id: 2147483647 rack: null) dead for group connect-radar-hdfs-sink-android-new (org.apache.kafka.clients.consumer.internals.AbstractCoordinator:565)
[2016-11-17 15:35:32,468] INFO Marking the coordinator radar-test.thehyve.net:9092 (id: 2147483647 rack: null) dead for group connect-radar-hdfs-sink-android-new (org.apache.kafka.clients.consumer.internals.AbstractCoordinator:565)
[2016-11-17 15:35:32,468] ERROR Task radar-hdfs-sink-android-new-1 threw an uncaught and unrecoverable exception (org.apache.kafka.connect.runtime.WorkerTask:142)
java.lang.NullPointerException
    at org.apache.kafka.clients.consumer.internals.ConsumerNetworkClient.poll(ConsumerNetworkClient.java:162)
    at org.apache.kafka.clients.consumer.internals.AbstractCoordinator.ensureCoordinatorReady(AbstractCoordinator.java:181)
    at org.apache.kafka.clients.consumer.internals.AbstractCoordinator.ensureActiveGroup(AbstractCoordinator.java:229)
    at org.apache.kafka.clients.consumer.internals.ConsumerCoordinator.ensurePartitionAssignment(ConsumerCoordinator.java:366)
    at org.apache.kafka.clients.consumer.KafkaConsumer.pollOnce(KafkaConsumer.java:975)
    at org.apache.kafka.clients.consumer.KafkaConsumer.poll(KafkaConsumer.java:938)
    at org.apache.kafka.connect.runtime.WorkerSinkTask.pollConsumer(WorkerSinkTask.java:316)
    at org.apache.kafka.connect.runtime.WorkerSinkTask.poll(WorkerSinkTask.java:222)
    at org.apache.kafka.connect.runtime.WorkerSinkTask.iteration(WorkerSinkTask.java:170)
    at org.apache.kafka.connect.runtime.WorkerSinkTask.execute(WorkerSinkTask.java:142)
    at org.apache.kafka.connect.runtime.WorkerTask.doRun(WorkerTask.java:140)
    at org.apache.kafka.connect.runtime.WorkerTask.run(WorkerTask.java:175)
    at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
    at java.util.concurrent.FutureTask.run(FutureTask.java:266)
    at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
    at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
    at java.lang.Thread.run(Thread.java:745)
[2016-11-17 15:35:32,469] ERROR Task is being killed and will not recover until manually restarted (org.apache.kafka.connect.runtime.WorkerTask:143)
[2016-11-17 15:35:32,467] INFO Marking the coordinator radar-test.thehyve.net:9092 (id: 2147483647 rack: null) dead for group connect-radar-hdfs-sink-android-test (org.apache.kafka.clients.consumer.internals.AbstractCoordinator:565)
[2016-11-17 15:35:32,469] INFO Marking the coordinator radar-test.thehyve.net:9092 (id: 2147483647 rack: null) dead for group connect-radar-hdfs-sink-android-test (org.apache.kafka.clients.consumer.internals.AbstractCoordinator:565)
[2016-11-17 15:35:32,477] ERROR Task radar-hdfs-sink-android-test-1 threw an uncaught and unrecoverable exception (org.apache.kafka.connect.runtime.WorkerTask:142)
java.lang.NullPointerException
    at org.apache.kafka.clients.consumer.internals.ConsumerNetworkClient.poll(ConsumerNetworkClient.java:162)
    at org.apache.kafka.clients.consumer.internals.AbstractCoordinator.ensureCoordinatorReady(AbstractCoordinator.java:181)
    at org.apache.kafka.clients.consumer.internals.AbstractCoordinator.ensureActiveGroup(AbstractCoordinator.java:229)
    at org.apache.kafka.clients.consumer.internals.ConsumerCoordinator.ensurePartitionAssignment(ConsumerCoordinator.java:366)
    at org.apache.kafka.clients.consumer.KafkaConsumer.pollOnce(KafkaConsumer.java:975)
    at org.apache.kafka.clients.consumer.KafkaConsumer.poll(KafkaConsumer.java:938)
    at org.apache.kafka.connect.runtime.WorkerSinkTask.pollConsumer(WorkerSinkTask.java:316)
    at org.apache.kafka.connect.runtime.WorkerSinkTask.poll(WorkerSinkTask.java:222)
    at org.apache.kafka.connect.runtime.WorkerSinkTask.iteration(WorkerSinkTask.java:170)
    at org.apache.kafka.connect.runtime.WorkerSinkTask.execute(WorkerSinkTask.java:142)
    at org.apache.kafka.connect.runtime.WorkerTask.doRun(WorkerTask.java:140)
    at org.apache.kafka.connect.runtime.WorkerTask.run(WorkerTask.java:175)
    at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
    at java.util.concurrent.FutureTask.run(FutureTask.java:266)
    at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
    at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
    at java.lang.Thread.run(Thread.java:745)
[2016-11-17 15:35:32,478] ERROR Task is being killed and will not recover until manually restarted (org.apache.kafka.connect.runtime.WorkerTask:143)
[2016-11-17 15:35:32,469] INFO Marking the coordinator radar-test.thehyve.net:9092 (id: 2147483647 rack: null) dead for group connect-radar-hdfs-sink-android-new (org.apache.kafka.clients.consumer.internals.AbstractCoordinator:565)
[2016-11-17 15:35:32,479] ERROR Task radar-hdfs-sink-android-new-2 threw an uncaught and unrecoverable exception (org.apache.kafka.connect.runtime.WorkerTask:142)
```
What is the actual route cause of the issue?
Should i configure more properties?

**Comment by cotedm** [comment]
@nivemaham I wonder if this has been fixed in a later version of Kafka. There's nothing here to suggest that this is specific to the HDFS Sink Connector, but rather a bug in the Consumer. It seems possible that this was fixed with the release of 0.10.1 and the fix to some race condition on the completion of futures [here](https://github.com/apache/kafka/pull/1821)? It would be good to know what version you are running and find out if it occurs on 0.10.1 to see if it's worth opening a JIRA for.

**Comment by cotedm** [comment]
Closing this one as I believe it's fixed in the latest release of Kafka and doesn't appear to be a problem specifically with this connector.
